pkg/virt-handler/vm.go

```diff
@@ -457,11 +457,19 @@ func (d *VirtualMachineController) execute(key string) error {
 }
 
 func (d *VirtualMachineController) replacePVCByHostDisk(vmi *v1.VirtualMachineInstance) error {
-	// if PVC is defined then it is replaced by HostDisk
-	// PersistenVolumeClaim is mounted into pod as directory from node filesystem
-	// it is not attached as independent block device
+	// if PVC is defined and it's not a BlockMode PVC, then it is replaced by HostDisk
+	// Filesystem PersistenVolumeClaim is mounted into pod as directory from node filesystem
 	for i := range vmi.Spec.Volumes {
 		if volumeSource := &vmi.Spec.Volumes[i].VolumeSource; volumeSource.PersistentVolumeClaim != nil {
+
+			isBlockVolumePVC, err := hostdisk.IsBlockVolumePVC(volumeSource.PersistentVolumeClaim.ClaimName, vmi.Namespace, d.clientset)
+			if err != nil {
+				return err
+			}
+			if isBlockVolumePVC {
+				continue
```

Comment:
If the ```BlockVolumePVC``` is empty then virt-launcher will not create a disk.img, a user will have to do it by himself.
Instead of skipping, maybe it's a good idea to get a real size of PVC (not over API but https://github.com/kubevirt/kubevirt/pull/1349#discussion_r205513560) and let it be converted to HostDisk?

Reply:
I'm not sure we can introspect the contents of the volume in this case. Since it's block mode, it's not mounted/we can't tell if it's empty or not.

Reply:
Yeah a VMI can always write to the block-device without initialization from our side. The block-device is the "empty disk" already in this case.

Reply:
Ok, understood.